enhancements/agent-installer/development-environment.md

```diff
@@ -0,0 +1,305 @@
+* Re-factoring parts of dev-scripts that are needed for this ephemeral agent based installer.
+* Convergence of the underlying scripts and environment setup code between Metal Platform (IPI) 
+team and the new Agent Installer teams.
+
+### Non-Goals
+
+* Re-factoring dev-scripts and assisted-test-infra for reasons not related to the new 
+ephemeral agent based installer image project.
+* Overall convergence between all the installer development teams of scripts and e2e test flows.
+
+## Proposal
+
+Create an development environment for the ephemeral agent based installer image in the 
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts) repository and refactor 
+components as needed for the new project.
+
+### User Stories
+
+* As a developer, I need to set up an environment so that I can simulate the installation of a 
+users first cluster using the ephemeral based agent installer image.
+
+* As a developer, I need to launch the built ephemeral agent based installer image so that I can test 
+and validate this image during development.
+
+* As a developer, I need to install Openshift using the ephemeral agent based installer image 
+to validate the agent installer.
+
+* As a developer, I need to check cluster installation progress of the Openshift cluster being
+installed by agent installer and when this installation is complete to validate the Openshift
+cluster.
+
+* As a developer, I need to validate the kubeconfig is available so I can use it to access the
+installed cluster.
+
+* As a developer, I want to extract logs from agent installer and the Openshift cluster under
+install to troubleshoot and debug.
+
+### API Extensions
+
+Not applicable.
+
+### Risks and Mitigations
+
+The direction of this proposal is now to extend and refactor parts of dev-scripts
+for the agent-based installer.
+
+** There will be now a third team using dev-scripts for slightly different purposes.
+Which means the risk of breaking things in the central framework now is larger and
+has a larger "splash zone."
+
+We can mitigate this for the new ephemeral agent installer by adding e2e tests into the
+dev-scripts repository that run the agent flows. The purpose of this would be to find 
+breakages in the tests across the different teams.
+
+** The dev-scripts repository clones the [metal-dev-env](https://github.com/metal3-io/metal3-dev-env) 
+repository for most of its Ansible roles. Which means there is a risk of changes in this 
+repository can break things in our repository as well. We have less control over this repository 
+than others inside Openshift.
+
+This is partially mitigated already because the repository is cloned at a specific
+Git SHA. To further this we can add a test in dev-scripts that runs maybe once or twice
+a week that clones the HEAD of this repository. We would be alerted for breakages more
+frequently and we can mitigate them on a case-by-case basis.
```

Comment:
So far we haven't found this additional testing to be necessary, but we do sometimes have to make changes to dev-scripts when moving the metal3-dev-env pin (for example increasing ansible/python version)

Reply:
Let's not sign up for more work than we need. I like the idea of keeping this as a suggestion, but not a commitment of something we'll do.

Reply:
Okay I'll pair this one down to be a suggestion.